### ISSUE TITLE
How many epochs to train to get an successful result?

### ISSUE THREAD

**Issue by SeungyounShin** [lucastabelini/PolyLaneNet#23]
I trained tusimple_efficientnet-b0 (pretrained) for 40 epoches and loss are around 15. (I also got poor qualitative result from test.py)

**Comment by lucastabelini** [comment]
Depends on the dataset. For the paper, I trained on ELAS and LLAMAS for additional 385 and 75 epochs, respectively.

**Comment by SeungyounShin** [comment]
How about TuSimple? I trained TuSimple for 115epoches with efficientdet-b0, batch10
Accuracy is around 0.45 and qualitative result doesn't seems like it would be.
Default epoches is 2695 and I wonder how many epoches to train to get reasonable results.
Because, training 115 epoch has no advance since loss down to 15.

**Comment by lucastabelini** [comment]
>  Default epoches is 2695 and I wonder how many epoches to train to get reasonable results.

I can't remember what was the lowest number I did experiments with. You'll have to try a few different values if you do want to reduce it.

> Because, training 115 epoch has no advance since loss down to 15

With the Cosine Annealing learning rate scheduler (the default), it's important to adjust its parameters every time you modify the number of epochs. Therefore, make sure you did adjust it accordingly.
